### PR TITLE
Reworked components as classes and started on integrating the Board 

### PR DESCRIPTION
**Pull request by gleason9113**
I started digging into the and ended up reworking App.js and adding a couple of components.
The first is the addition of a socket component which if I understand what I've been reading (iffy) should eliminate the possibility of extra sockets during a session.
This appears to work based on what I've been able to test so far.

I've run into two issues with the code as is, and I'm reasonably sure it's bc I'm tired and I've been looking at it too long.  
The app will correctly flip between the login component and the Board based on the state of the component (Currently a variable named 'connected' is the flag for that).  The problem I've run into is that I can't get the form data to update the component state, and after extensive debugging I now can't seem to get the form data from the event at all.  I'm going to bed, but will be grateful for any input!

### CODE REVIEW

**Comment by sashetov** [comment]
I'm trying to review this change but too many things have changed.
I do see the socket set to null. maybe you should use the previous method that seemed to work, with useState and useEffect

https://beta.reactjs.org/apis/react/useState <- lets you add a state variable to your component
https://reactjs.org/docs/hooks-reference.html#useeffect <- lets you  run effects after every completed render. The function passed to useEffect will run after the render is committed to the screen. 

i think the method you are using may work but you may need to have some sort of post-render hander. it looks like these hooks are an effective way of doing. combining the two means you can get a socket after everything rendered and assign it to a state variable, the change of which will cause a conditional rerendering of the component if you made it work that way

**Comment by gleason9113** [comment]
Fixed the socket issue- login page now correctly flips to board on submit (I think- needs more testing, fer sher).
Todo:  
Get playerName from form and add to submission.
Get 2nd player joined to 1st players game.

**Comment by C3ra906** [comment]
Hi Ariel, I'm currently getting a blank screen on Chrome when I run the program using npm run build-react and npm start

File: src/components/socket.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React, { createContext } from 'react';
+import { io, Socket } from 'socket.io-client';
+const url = ('http://localhost:8080');
+
+const socket = io(url), SocketContext = createContext(socket);
```
Comment (sashetov) [inline src/components/socket.js:5]: when you open the page in the browser there is an error in console:

TypeError: (0 , c.io) is not a function
    at socket.js:5:16

see line 5 for that. you may want to try io.connect or something

**Review by sashetov — CHANGES_REQUESTED** [review]

**Comment by gleason9113** [comment]
Hmm.  I'm not sure how to address that- it's loading as described when I run it.  The error you mentioned was something that I fixed earlier today- I pushed those changes to this branch this afternoon.

**Comment by sashetov** [comment]
the issue exists, here's a screenshot:
see screenshot

**Comment by gleason9113** [comment]
I believe you guys.  I just don't know why it's not not working on my machines.  I'm going to close this pull request, do a fresh push, and create a new one.

**Pull request closed without merging**